cirq/circuits/circuit.py

```diff
@@ -238,6 +238,37 @@ def __rmul__(self, repetitions: int):
             return NotImplemented
         return self * repetitions
 
+    def __pow__(self, exponent: float):
+        """See `cirq.pow`.
+
+        For integer exponents, this corresponds to repetitions of the moments
+        of the circuit. When the exponent is negative, these circuits have
+        their moment order reversed and the inverse of each operation in the
+        moment is applied. This means circuit**-1 corresponds to the inverse
+        circuit.
+
+        For non-integer exponents, the integer closest to zero is calculated
```

Comment:
Do we need the non-integer case?

I'm fine with defining a convention for raising moments to a fractional power. But it's odd that the convention is different from the convention used elsewhere in the library, which is that raising to a power maps directly onto the eigenvalues of a thing. For example, this definition is inconsistent with the definition we're using for `XXPowGate` meaning`circuit((X1 * X2)**t) != circuit((X1 * X2))**t`.

Reply:
Hmmm, that is another interesting point; that we already define multiplication to be repetition.

I agree that getting the inverse to work is good. Integer powers are a bit ambiguous w.r.t. multiplication but probably worth it. I think we should hold off on fractional powers.